jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/bucketEditor/S3ViewerPanel.java

```diff
@@ -111,107 +128,113 @@ public boolean isCellEditable(Object o) {
                 }
             };
 
-            ColumnInfo eTag = new ColumnInfo<Object, Object>("Etag") {
+            ColumnInfo size = new ColumnInfo<Object, Object>("Size") {
                 @Override
                 @Nullable
                 public Object valueOf(Object obj) {
-                    VirtualFile file = getVirtualFile(obj);
-                    return (file instanceof S3VirtualFile) ? ((S3VirtualFile) file).getFile().getETag() : "";
+                    VirtualFile file = getVirtualFileFromNode(obj);
+                    return (file instanceof S3VirtualFile) ?
+                            ((S3VirtualFile) file).formatSize() : "";
                 }
 
                 public boolean isCellEditable(Object o) {
                     return true;
                 }
+
+
             };
 
-            ColumnInfo size = new ColumnInfo<Object, Object>("Size") {
+            ColumnInfo modified = new ColumnInfo<Object, Object>("Last-Modified") {
                 @Override
                 @Nullable
                 public Object valueOf(Object obj) {
-                    VirtualFile file = getVirtualFile(obj);
-                    return (file instanceof S3VirtualFile) ?
-                            ((S3VirtualFile) file).formatSize() : "";
+                    VirtualFile file = getVirtualFileFromNode(obj);
+                    return (file instanceof S3VirtualFile) ? ((S3VirtualFile) file).formatDate(
+                            ((S3VirtualFile) file).getFile().getLastModified()) : "";
                 }
 
                 public boolean isCellEditable(Object o) {
                     return true;
                 }
             };
 
-            ColumnInfo modified = new ColumnInfo<Object, Object>("Last-Modified") {
+            ColumnInfo eTag = new ColumnInfo<Object, Object>("Etag") {
                 @Override
                 @Nullable
                 public Object valueOf(Object obj) {
-                    VirtualFile file = getVirtualFile(obj);
-                    return (file instanceof S3VirtualFile) ? ((S3VirtualFile) file).formatDate(
-                            ((S3VirtualFile) file).getFile().getLastModified()) : "";
+                    VirtualFile file = getVirtualFileFromNode(obj);
```

Comment:
The use of `ColumnInfo` is a little repetitive, can we generalize this?

Reply:
I tried cutting down some of the logic by using `getVirtualFileFromNode(obj)`. However, I am not completely sure if I can generalize this - since every column needs to have a unique `valueOf`.

Reply:
maybe something like:

```kotlin
// I feel like we can do better than <Any, Any>
class S3ColumnInfo(columnTitle: String, private val valueGetter: (S3VirtualFile) -> String?) :
ColumnInfo<Any, Any>(columnTitle) {
    override fun valueOf(obj: Any?): Any? {
        val file = getVirtualFileFromNode(obj)
        return if (file is S3VirtualFile) {
            valueGetter.invoke(file)
        } else {
            ""
        }
    }
}

// java
ColumnInfo eTag = new S3ColumnInfo(
    "Etag",
    virtualFile -> virtualFile.getFile().getETag().replace("\"", "")
);
```
?

Reply:
If you wanted to do the above class in Java, Kotlin lambdas are roughly analogous to [java.util.function](https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html).

Reply:
This works out perferctly. Thanks. Will update.